src/orquestra/sdk/_ray/_build_workflow.py

```diff
@@ -214,7 +214,14 @@ def _ray_remote(*inner_args, **inner_kwargs):
                     deserialize=serialization,
                 )
 
-                wrapped_return = wrapped(*inner_args, **inner_kwargs)
+                try:
+                    wrapped_return = wrapped(*inner_args, **inner_kwargs)
+                except SystemError as e:
+                    raise EnvironmentError(
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏`EnvrionmentError` is only available for backwards compatibility (since 3.3) and is an alias of `OSError`. Perhaps a custom exception may be a better thing to use.